exist-core/src/main/java/org/exist/xquery/functions/fn/FunSubSequence.java

```diff
@@ -65,21 +65,32 @@ public FunSubSequence(final XQueryContext context, final FunctionSignature signa
 
     @Override
     public void analyze(final AnalyzeContextInfo contextInfo) throws XPathException {
-        // statically check the argument list
-        checkArguments();
-        // call analyze for each argument
         inPredicate = (contextInfo.getFlags() & IN_PREDICATE) > 0;
         unordered = (contextInfo.getFlags() & UNORDERED) > 0;
         contextId = contextInfo.getContextId();
         contextInfo.setParent(this);
 
+        final SequenceType[] argumentTypes = mySignature.getArgumentTypes();
         for (int i = 0; i < getArgumentCount(); i++) {
+            final Expression arg = getArgument(i);
+
+            // call analyze for each argument
             final AnalyzeContextInfo argContextInfo = new AnalyzeContextInfo(contextInfo);
-            getArgument(i).analyze(argContextInfo);
+            arg.analyze(argContextInfo);
             if (i == 0) {
                 contextInfo.setStaticReturnType(argContextInfo.getStaticReturnType());
             }
+
+            if (!argumentsChecked) {
```

Comment:
This block appear three times in the changed files. Would it be possible to extract it into a method, function that can be re-used?

Reply:
It doesn't make sense as these are function specific (the override the default in `Function`), and actually I think it only appears twice the same, the other instance is slightly different.